src/detect-engine-iponly.c

```diff
@@ -883,25 +883,17 @@ void IPOnlyInit(DetectEngineCtx *de_ctx, DetectEngineIPOnlyCtx *io_ctx)
                                                   SigNumArrayPrint);
     io_ctx->tree_ipv6dst = SCRadixCreateRadixTree(SigNumArrayFree,
                                                   SigNumArrayPrint);
+
+    io_ctx->sig_mapping = SCCalloc(1, de_ctx->sig_array_len * sizeof(uint32_t));
+    io_ctx->sig_mapping_size = 0;
 }
 
-/**
- * \brief Setup the IP Only thread detection engine context
- *
- * \param de_ctx Pointer to the current detection engine
- * \param io_ctx Pointer to the current ip only thread detection engine
- */
-void DetectEngineIPOnlyThreadInit(DetectEngineCtx *de_ctx,
-                                  DetectEngineIPOnlyThreadCtx *io_tctx)
+SigIntId IPOnlyTrackSigNum(DetectEngineIPOnlyCtx *io_ctx, SigIntId signum)
 {
-    /* initialize the signature bitarray */
-    io_tctx->sig_match_size = de_ctx->io_ctx.max_idx / 8 + 1;
-    io_tctx->sig_match_array = SCMalloc(io_tctx->sig_match_size);
-    if (io_tctx->sig_match_array == NULL) {
-        exit(EXIT_FAILURE);
-    }
-
-    memset(io_tctx->sig_match_array, 0, io_tctx->sig_match_size);
+    SigIntId loc = io_ctx->sig_mapping_size;
+    io_ctx->sig_mapping[loc] = signum;
```

Comment:
could just be `io_ctx->sig_mapping[io_ctx->sig_mapping_size++] = signum;`?

Reply:
it could, but then it would have to

```
return io_ctx->sig_mapping_size-1
```
to return the index that was allocated.

Reply:
Oh, i see what you mean like this?

```
SigIntId loc = io_ctx->sig_mapping_size;
io_ctx->sig_mapping[io_ctx->sig_mapping_size++] = signum;
return loc;
```

I had split it up because I had it inlined the first time but it was slightly wrong and it was returning the wrong value.